### DISCUSSION TITLE
Is it possible to have an object for value?

### DISCUSSION THREAD

**Discussion by magick93** [svecosystem/formsnap#122] [Q&A]
Hello

I'm working with a schema like this:

```
export const numberRange = z.object({
    lower: z.number(),
    upper: z.number()
}).refine(data => data.lower <= data.upper, {
    message: "The lower bound must be less than or equal to the upper bound"
});
export type NumberRange = z.infer<typeof numberRange>;

export const business_profile = z.object({
    industry: z.string().optional(),
    company_size: numberRange.optional(),
    business_model_type: z.string().optional(),
});
```

I'm wondering how I can assign `numberRange` to the value for a form item. 

Is it currently possible?

Thanks

**Comment by ambergristle** [comment]
formsnap is built on superforms, which generally equates fields with single values. depending on your use-case, you may find it helpful to use proxies to access state, or simply to access nested fields by dot-path:
- proxies: https://superforms.rocks/concepts/proxy-objects
- dot-path: 
```
<Field {form} name="company_size.lower" />
<Field {form} name="company_size.upper" />
```
as the values are related, it may also be worth taking a look at fieldsets: https://www.formsnap.dev/docs/components/fieldset
